src/Enzo/gravity/EnzoPhysicsGravity.hpp

```diff
@@ -0,0 +1,152 @@
+
+/// @file     enzo_EnzoPhysicsFluidProps.hpp
+/// @author   Matthew Abruzzo (matthewabruzzo@gmail.com)
+/// @date     2023-07-31
+/// @brief    [\ref Enzo] Declaration of the EnzoPhysicsGravity class
+
+#ifndef ENZO_GRAVITY_ENZO_PHYSICS_GRAVITY_HPP
+#define ENZO_GRAVITY_ENZO_PHYSICS_GRAVITY_HPP
+
+class EnzoPhysicsGravity : public Physics {
+
+  /// @class    EnzoPhysicsGravity
+  /// @ingroup  Enzo
+  /// @brief    [\ref Enzo] Provides a central interface for querying the
+  ///    gravitational constant.
+  ///
+  /// Handling of the gravitational constant depends on simulation's nature:
+  ///  - In non-cosmological sims the constant can be set to an arbitrary
+  ///    positive value (this helps simplify some test-problems).
+  ///  - Users aren't allow to specify the constant's value in cosmological
+  ///    simulations. This is because cosmological units are defined such that
+  ///    ``4*pi*G*rho_bar`` has the value ``1.0``, where ``rho_bar`` is the
+  ///    mean physical matter density of the universe.
+  ///
+  /// It seems like overkill to dedicate a whole class to storing a single
+  /// piece of information like the gravitational constant.
+  ///  - However, this information DEFINITELY belongs in a Physics class since
+  ///    it needs to be accessed in many (mostly-independent) Method classes
+  ///  - More importantly, there isn't a single Method class that can store
+  ///    this information. We will soon have 2 Methods that implement distinct
+  ///    methods for modelling self-gravity. Plus, it's possible to use a
+  ///    Method that implements a background potential independently of either
+  ///    of those choices.
+  ///  - In the future, it might make sense to dedicate a single Physics object
+  ///    to group miscellaneous physical values like the Gravitational constant
+  ///
+  /// @note
+  /// Some imporant notes include:
+  ///  - instances of this class can be initialized even if Gravity is not
+  ///    considered within a given simulation.
+  ///  - the data stored in the class is meant to be considered immutable.
+  ///  - enzo::cosmology() should return a consistent value over the lifetime
+  ///    of this class (in other words, EnzoPhysicsCosmology must be
+  ///    initialized first)
+
+public: // interface
+
+  /// Constructor
+  ///
+  /// @param grav_const_code_units When positive, this specifies the
+  ///     gravitational constant in code units. Otherwise, the gravitational
+  ///     constant is defined such that it's equal to the value of
+  //      enzo_constants::standard_grav_const when converted to cgs
+  EnzoPhysicsGravity(double grav_const_code_units)
+    : Physics()
+  {
+    if (grav_const_code_units <= 0.0) {
+      grav_constant_codeU_ = -1.0;
+    } else {
+      ASSERT("EnzoPhysicsGravity::EnzoPhysicsGravity",
+             "users aren't allowed to specify the gravitational constant in "
+             "a cosmological simulation", enzo::cosmology() == nullptr);
```

Comment:
Should this be !=; that is, throw the error if the `enzo::cosmology()` pointer was initialized?

Reply:
No this is correct as it's written. This is an assertion that `enzo::cosmology() == nullptr`. The error message is printed if that condition is violated.

I vaguely recall that I had things swapped around when I originally wrote this snippet (and I caught this mistake from a failing test)